orchestrator/api/api_v1/endpoints/subscriptions.py

```diff
@@ -120,6 +121,19 @@ def in_use_by_subscriptions(subscription_id: UUID) -> List[SubscriptionTable]:
     return query_in_use_by_subscriptions(subscription_id).all()
 
 
+@router.post("/subscriptions_for_in_used_by_ids", response_model=Dict[UUID, SubscriptionSchema])
+def subscriptions_by_in_used_by_ids(data: List[UUID] = Body(...)) -> Dict[UUID, SubscriptionSchema]:
+    subscriptions = {}
+    for id in data:
+        subscription_instance: SubscriptionInstanceTable = SubscriptionInstanceTable.query.get(id)
+        if subscription_instance:
+            subscription = SubscriptionTable.query.get(subscription_instance.subscription_id)
```

Comment:
Would it be possible to do this in 1 instead of 2*N queries?

Reply:
This seems to work

```python
   res = db.session.execute(
        select(SubscriptionInstanceTable.subscription_instance_id, SubscriptionTable)
        .filter(
            SubscriptionTable.subscription_id.in_(
                select(SubscriptionInstanceTable.subscription_id)
                .filter(SubscriptionInstanceTable.subscription_instance_id.in_(data))
                .subquery()
            )
        )

    ).all()
    return {row[0]: row[1] for row in res}
```

Only thing this lacks is the errors for missing instances, but I don't think this matters

Reply:
Probably yes. I was interested in the error because I had some fubar subscriptions
I have a test case for a wrong instance_id: so will try this and adapt the test.

Reply:
```/Users/acidjunk/surfnet/orchestrator/venv/lib/python3.10/site-packages/orchestrator/api/api_v1/endpoints/subscriptions.py:129: SAWarning: Coercing Subquery object into a select() for use in IN(); please pass a select() construct explicitly
  SubscriptionTable.subscription_id.in_(
/Users/acidjunk/surfnet/orchestrator/venv/lib/python3.10/site-packages/orchestrator/api/api_v1/endpoints/subscriptions.py:127: SAWarning: SELECT statement has a cartesian product between FROM element(s) "subscriptions" and FROM element "subscription_instances".  Apply join condition(s) between each element to resolve.
  res = db.session.execute(
```

The result would also contain a lot of ID's that not originated from the data array.

Reply:
Tried it a bit more:
https://github.com/workfloworchestrator/orchestrator-core/pull/131